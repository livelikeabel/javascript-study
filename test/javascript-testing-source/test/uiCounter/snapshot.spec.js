import $ from 'jquery';
import {createUICounter} from '../../src/uiCounter/uiCounter';

let $el;

beforeEach(() => {
  $el = $('<div>');
  $el.appendTo(document.body);

  createUICounter($el, {
    initVal: 10,
    min: 8,
    max: 12
  });
});

afterEach(() => {
  document.body.innerHTML = '';
});

it('생성시 버튼과 초기값을 렌더링한다.', () => {
  expect($el.html()).toMatchSnapshot();
});

it('+ 버튼 클릭시 1 증가한다.', () => {
  $el.find('.btn-inc').click();

  expect($el.html()).toMatchSnapshot();
});

it('Max값인 경우 + 버튼이 disabled 상태가 되며 클릭해도 증가하지 않는다.', () => {
  $el.find('.btn-inc').click();
  $el.find('.btn-inc').click();
  $el.find('.btn-inc').click();

  expect($el.html()).toMatchSnapshot();
});

it('- 버튼 클릭시 1 감소한다.', () => {
  $el.find('.btn-dec').click();

  expect($el.html()).toMatchSnapshot();
});

it('Min값인 경우 - 버튼이 disabled 상태가 되며, 클릭해도 감소하지 않는다.', () => {
  $el.find('.btn-dec').click();
  $el.find('.btn-dec').click();
  $el.find('.btn-dec').click();

  expect($el.html()).toMatchSnapshot();
});
