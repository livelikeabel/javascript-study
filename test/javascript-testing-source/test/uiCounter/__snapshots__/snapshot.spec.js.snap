// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`+ 버튼 클릭시 1 증가한다. 1`] = `
"
      <button type=\\"button\\" class=\\"btn btn-secondary btn-dec\\">-</button>
      <span class=\\"value\\">11</span>
      <button type=\\"button\\" class=\\"btn btn-primary btn-inc\\">+</button>
    "
`;

exports[`- 버튼 클릭시 1 감소한다. 1`] = `
"
      <button type=\\"button\\" class=\\"btn btn-secondary btn-dec\\">-</button>
      <span class=\\"value\\">9</span>
      <button type=\\"button\\" class=\\"btn btn-primary btn-inc\\">+</button>
    "
`;

exports[`Max값인 경우 + 버튼이 disabled 상태가 되며 클릭해도 증가하지 않는다. 1`] = `
"
      <button type=\\"button\\" class=\\"btn btn-secondary btn-dec\\">-</button>
      <span class=\\"value\\">12</span>
      <button type=\\"button\\" disabled=\\"\\" class=\\"btn btn-primary btn-inc\\">+</button>
    "
`;

exports[`Min값인 경우 - 버튼이 disabled 상태가 되며, 클릭해도 감소하지 않는다. 1`] = `
"
      <button type=\\"button\\" disabled=\\"\\" class=\\"btn btn-secondary btn-dec\\">-</button>
      <span class=\\"value\\">8</span>
      <button type=\\"button\\" class=\\"btn btn-primary btn-inc\\">+</button>
    "
`;

exports[`생성시 버튼과 초기값을 렌더링한다. 1`] = `
"
      <button type=\\"button\\" class=\\"btn btn-secondary btn-dec\\">-</button>
      <span class=\\"value\\">10</span>
      <button type=\\"button\\" class=\\"btn btn-primary btn-inc\\">+</button>
    "
`;
